bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/handler/SoulissGatewayHandler.java

```diff
@@ -0,0 +1,331 @@
+ * @author Luca Calcaterra - Refactor for OH3
+ */
+@NonNullByDefault
+public class SoulissGatewayHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(SoulissGatewayHandler.class);
+    public @Nullable DatagramSocket datagramSocketDefaultPort;
+    private @Nullable SoulissBindingUDPServerJob udpServerDefaultPortRunnableClass;
+    private SoulissCommonCommands soulissCommands = new SoulissCommonCommands();
+
+    boolean bGatewayDetected = false;
+
+    private @NonNullByDefault({}) Configuration gwConfigurationMap;
+
+    public int pingRefreshInterval;
+    public int subscriptionRefreshInterval;
+    public boolean thereIsAThingDetection = true;
+    public int healthRefreshInterval;
+    public int sendRefreshInterval;
+    public int sendTimeoutToRequeue;
+    public int sendTimeoutToRemovePacket;
+    private Bridge bridge;
+    public int preferredLocalPort;
+    public int soulissGatewayPort;
+    public byte userIndex;
+    public byte nodeIndex;
+    public String ipAddressOnLAN = "";
+    private int nodes;
+    private int maxTypicalXnode;
+    private int countPingKo = 0;
+    // private int maxnodes = 0;
+    // private int maxrequests = 0;
+
+    public SoulissGatewayHandler(Bridge _bridge) {
+        super(_bridge);
+        bridge = _bridge;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initializing server handler for thingGeneric {}", getThing());
+
+        gwConfigurationMap = bridge.getConfiguration();
+        ipAddressOnLAN = (String) gwConfigurationMap.get(SoulissBindingConstants.CONFIG_IP_ADDRESS);
+        if (gwConfigurationMap == null) {
+            logger.debug("Gateway Handler - Error in Configuration Map - Thing= {}", getThing());
+        } else {
+            if (gwConfigurationMap.get(SoulissBindingConstants.CONFIG_LOCAL_PORT) != null) {
+                preferredLocalPort = ((BigDecimal) gwConfigurationMap.get(SoulissBindingConstants.CONFIG_LOCAL_PORT))
+                        .intValue();
+                logger.debug("Get Preferred Local Port: {}", preferredLocalPort);
+            }
+
+            if (preferredLocalPort < 0 && preferredLocalPort > 65000) {
+                bridge.getConfiguration().put(SoulissBindingConstants.CONFIG_LOCAL_PORT, 0);
+                logger.debug("Set Preferred Local Port to {}", 0);
+            }
+
+            if (gwConfigurationMap.get(SoulissBindingConstants.CONFIG_PORT) != null) {
+                soulissGatewayPort = ((BigDecimal) gwConfigurationMap.get(SoulissBindingConstants.CONFIG_PORT))
+                        .intValue();
+                logger.debug("Get Souliss Gateway Port: {}", soulissGatewayPort);
+            }
+            if (soulissGatewayPort < 0 && soulissGatewayPort > 65000)
+
+            {
+                bridge.getConfiguration().put(SoulissBindingConstants.CONFIG_PORT,
```

Comment:
This is an example of not using gwConfigurationMap and getting it a second time. The example of using a config class will make your code cleaner, shorter, faster and easier to maintain. Recommend you look at that method instead.

Reply:
idem as above

Reply:
done